presto-main/src/main/java/com/facebook/presto/sql/planner/plan/SemiJoinNode.java

```diff
@@ -103,6 +114,18 @@ public Symbol getSemiJoinOutput()
         return filteringSourceHashSymbol;
     }
 
+    @JsonProperty("method")
+    public Optional<Method> getMethod()
+    {
+        return method;
+    }
+
+    public boolean isDistributed()
+    {
+        checkState(method.isPresent(), "method not yet set");
```

Comment:
Is it ever possible that join node is created with an empty optional? There should be a default behavior.